packages/components/src/components/hds/app-side-nav/index.ts

```diff
@@ -0,0 +1,195 @@
+    isResponsive?: boolean;
+    isCollapsible?: boolean;
+    isMinimized?: boolean;
+    ariaLabel?: string;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    onToggleMinimizedStatus?: (arg: boolean) => void;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    onDesktopViewportChange?: (arg: boolean) => void;
+  };
+  Blocks: {
+    header?: [
+      {
+        Header?: HdsAppSideNavBaseSignature['Blocks']['header'];
+        isMinimized?: boolean;
+      },
+    ];
+    body?: [
+      {
+        Body?: HdsAppSideNavBaseSignature['Blocks']['body'];
+        isMinimized?: boolean;
+      },
+    ];
+    footer?: [
+      {
+        Footer?: HdsAppSideNavBaseSignature['Blocks']['footer'];
+        isMinimized?: boolean;
+      },
+    ];
+  };
+  Element: HdsAppSideNavBaseSignature['Element'];
+}
+
+export default class HdsAppSideNavComponent extends Component<HdsAppSideNavSignature> {
+  @tracked isResponsive = this.args.isResponsive ?? true; // controls if the component reacts to viewport changes
+  @tracked isMinimized = this.args.isMinimized ?? false; // sets the default state on 'desktop' viewports
+  @tracked isCollapsible = this.args.isCollapsible ?? false; // controls if users can collapse the sidenav on 'desktop' viewports
+  @tracked isAnimating = false;
+  @tracked isDesktop = true;
+  desktopMQ: MediaQueryList;
+  containersToHide!: NodeListOf<Element>;
+
+  desktopMQVal = getComputedStyle(document.documentElement).getPropertyValue(
+    '--hds-app-desktop-breakpoint'
+  );
+
+  constructor(owner: unknown, args: HdsAppSideNavSignature['Args']) {
+    super(owner, args);
+    this.desktopMQ = window.matchMedia(`(min-width:${this.desktopMQVal})`);
+    this.addEventListeners();
+    registerDestructor(this, (): void => {
+      this.removeEventListeners();
+    });
+  }
+
+  addEventListeners(): void {
+    document.addEventListener('keydown', this.escapePress, true);
+    this.desktopMQ.addEventListener('change', this.updateDesktopVariable, true);
+    // if not instantiated as minimized via arguments
+    if (!this.args.isMinimized) {
+      // set initial state based on viewport using a "synthetic" event
+      const syntheticEvent = new MediaQueryListEvent('change', {
+        matches: this.desktopMQ.matches,
+        media: this.desktopMQ.media,
+      });
+      this.updateDesktopVariable(syntheticEvent);
+    }
+  }
+
+  removeEventListeners(): void {
+    document.removeEventListener('keydown', this.escapePress, true);
+    this.desktopMQ.removeEventListener(
+      'change',
+      this.updateDesktopVariable,
+      true
+    );
+  }
+
+  get shouldTrapFocus(): boolean {
+    return this.isResponsive && !this.isDesktop && !this.isMinimized;
+  }
+
+  get showToggleButton(): boolean {
+    return (this.isResponsive && !this.isDesktop) || this.isCollapsible;
+  }
+
+  /**
+   * @param ariaLabel
+   * @type {string}
+   * @default 'close menu'
+   */
+  get ariaLabel(): string {
```

Comment:
Nit: maybe this could be `toggleButtonAriaLabel` just so its clearer what this is for.